src/fdcache.h

```diff
@@ -81,24 +83,28 @@ class PageList
 
   private:
     void Clear(void);
-    bool Compress(void);
+    bool Compress(bool force_modified = false);
     bool Parse(off_t new_pos);
+    bool RawGetUnloadPageList(fdpage_list_t& dlpages, off_t offset, off_t size);
 
   public:
     static void FreeList(fdpage_list_t& list);
 
-    explicit PageList(off_t size = 0, bool is_loaded = false);
+    explicit PageList(off_t size = 0, bool is_loaded = false, bool is_modified = false);
+    explicit PageList(const PageList& other);
     ~PageList();
 
-    bool Init(off_t size, bool is_loaded);
+    bool Init(off_t size, bool is_loaded, bool is_modified);
     off_t Size(void) const;
-    bool Resize(off_t size, bool is_loaded);
+    bool Resize(off_t size, bool is_loaded, bool is_modified);
 
     bool IsPageLoaded(off_t start = 0, off_t size = 0) const;                  // size=0 is checking to end of list
-    bool SetPageLoadedStatus(off_t start, off_t size, bool is_loaded = true, bool is_compress = true);
+    bool SetPageLoadedStatus(off_t start, off_t size, bool is_loaded = true, bool is_modified = false, bool is_compress = true);
```

Comment:
As you say, the symbols are easier to understand.
It is possible to make these into separate symbols, but since "modified" and "loaded" are independent, you will be preparing all the synthesized patterns.
In other words, it will be like a bitfield. (Can also be done with enum)
As a result, it may not be intuitive as it only complicates the expression.
The part that uses these functions is limited, can we recommend it as it is now?

Reply:
I'm more concerned about the existing default arguments changing meaning since the parameter list reordered.  If we keep the same order, then none of the existing call sites need to change orders.

Reply:
The loaded and modified flag is changed an enum.
However, we may need to refactor it later, but now we will make this change.